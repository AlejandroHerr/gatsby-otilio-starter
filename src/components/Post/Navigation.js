// @flow
import React from 'react';

import type { PreviewType } from '../../types/post';

import NavigationLink from './NavigationLink';

import styles from './Post.module.scss';

const Navigation = ({ next, prev }: {next ?: PreviewType, prev?: PreviewType}) => (
  <aside className={styles.post_navigation}>
    {next &&
      <NavigationLink
        className={styles.post_navigation__next}
        icon="chevron_left"
        {...next}
      />
    }
    {prev &&
      <NavigationLink
        className={styles.post_navigation__prev}
        icon="chevron_right"
        {...prev}
      />
    }
    <div className={styles.post_navigation__clear} />
    <div className={styles.post_navigation__separator} />
  </aside>
);

Navigation.defaultProps = {
  next: null,
  prev: null,
};

export default Navigation;
