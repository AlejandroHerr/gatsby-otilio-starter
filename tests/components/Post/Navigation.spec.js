import React from 'react';
import renderer from 'react-test-renderer';

import Navigation from '../../../src/components/Post/Navigation';

const next = {
  excerpt: 'prevExcerpt',
  path: 'prevPath',
  title: 'prevTitle',
};

const prev = {
  excerpt: 'prevExcerpt',
  path: 'prevPath',
  title: 'prevTitle',
};

describe('Post', () => {
  describe('Navigation', () => {
    it('should match the exact snapshot with prev and next', () => {
      const tree = renderer.create(<Navigation next={next} prev={prev} />).toJSON();

      expect(tree).toMatchSnapshot();
    });
    it('should match the exact snapshot with prev', () => {
      const tree = renderer.create(<Navigation prev={prev} />).toJSON();

      expect(tree).toMatchSnapshot();
    });
    it('should match the exact snapshot with next', () => {
      const tree = renderer.create(<Navigation next={next} />).toJSON();

      expect(tree).toMatchSnapshot();
    });
    it('should match the exact snapshot with none', () => {
      const tree = renderer.create(<Navigation />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
