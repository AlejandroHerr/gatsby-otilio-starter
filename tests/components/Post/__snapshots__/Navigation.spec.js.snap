// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Navigation should match the exact snapshot with next 1`] = `
<aside
  className="post_navigation"
>
  <a
    className="post_navigation__link post_navigation__next"
    to="prevPath"
  >
    <section
      className="post_navigation__teaser"
    >
      <div
        className="post_navigation__icon"
      >
        <svg
          className="byte_icon byte_icon__chevron_left"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path />
        </svg>
      </div>
      <h2
        className="post_navigation__title"
      >
        prevTitle
      </h2>
      <p
        className="post_navigation__excerpt"
      >
        prevExcerpt
      </p>
    </section>
  </a>
  <div
    className="post_navigation__clear"
  />
  <div
    className="post_navigation__separator"
  />
</aside>
`;

exports[`Post Navigation should match the exact snapshot with none 1`] = `
<aside
  className="post_navigation"
>
  <div
    className="post_navigation__clear"
  />
  <div
    className="post_navigation__separator"
  />
</aside>
`;

exports[`Post Navigation should match the exact snapshot with prev 1`] = `
<aside
  className="post_navigation"
>
  <a
    className="post_navigation__link post_navigation__prev"
    to="prevPath"
  >
    <section
      className="post_navigation__teaser"
    >
      <div
        className="post_navigation__icon"
      >
        <svg
          className="byte_icon byte_icon__chevron_right"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path />
        </svg>
      </div>
      <h2
        className="post_navigation__title"
      >
        prevTitle
      </h2>
      <p
        className="post_navigation__excerpt"
      >
        prevExcerpt
      </p>
    </section>
  </a>
  <div
    className="post_navigation__clear"
  />
  <div
    className="post_navigation__separator"
  />
</aside>
`;

exports[`Post Navigation should match the exact snapshot with prev and next 1`] = `
<aside
  className="post_navigation"
>
  <a
    className="post_navigation__link post_navigation__next"
    to="prevPath"
  >
    <section
      className="post_navigation__teaser"
    >
      <div
        className="post_navigation__icon"
      >
        <svg
          className="byte_icon byte_icon__chevron_left"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path />
        </svg>
      </div>
      <h2
        className="post_navigation__title"
      >
        prevTitle
      </h2>
      <p
        className="post_navigation__excerpt"
      >
        prevExcerpt
      </p>
    </section>
  </a>
  <a
    className="post_navigation__link post_navigation__prev"
    to="prevPath"
  >
    <section
      className="post_navigation__teaser"
    >
      <div
        className="post_navigation__icon"
      >
        <svg
          className="byte_icon byte_icon__chevron_right"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path />
        </svg>
      </div>
      <h2
        className="post_navigation__title"
      >
        prevTitle
      </h2>
      <p
        className="post_navigation__excerpt"
      >
        prevExcerpt
      </p>
    </section>
  </a>
  <div
    className="post_navigation__clear"
  />
  <div
    className="post_navigation__separator"
  />
</aside>
`;
